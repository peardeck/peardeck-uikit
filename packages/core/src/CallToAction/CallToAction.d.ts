import { ReactNode } from "react";
interface CallToActionPropTypes {
  children: ReactNode;
  title: ReactNode;
  image: ReactNode;
  buttonContent: ReactNode;
}
export declare const CallToAction: ({
  children,
  title,
  image,
  buttonContent,
}: CallToActionPropTypes) => JSX.Element;
export {};
