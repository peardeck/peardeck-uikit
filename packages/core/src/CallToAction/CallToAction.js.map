{"version":3,"file":"CallToAction.js","sourceRoot":"","sources":["CallToAction.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAoB,MAAM,OAAO,CAAC;AACzC,OAAO,MAAM,MAAM,mBAAmB,CAAC;AACvC,OAAO,EAAE,MAAM,EAAE,MAAM,kBAAkB,CAAC;AAE1C,MAAM,kBAAkB,GAAG,MAAM,CAAC,GAAG,CAAA;;;;;;;;;;;;;;;;CAgBpC,CAAC;AAEF,MAAM,kCAAkC,GAAG,MAAM,CAAC,GAAG,CAAA;;;;;;CAMpD,CAAC;AAEF,MAAM,yBAAyB,GAAG,MAAM,CAAC,CAAC,CAAA;;;CAGzC,CAAC;AAEF,MAAM,+BAA+B,GAAG,MAAM,CAAC,EAAE,CAAA;;CAEhD,CAAC;AAEF,MAAM,gCAAgC,GAAG,MAAM,CAAC,GAAG,CAAA;;;;CAIlD,CAAC;AASF,MAAM,CAAC,MAAM,YAAY,GAAG,CAAC,EAC3B,QAAQ,EACR,KAAK,EACL,KAAK,EACL,aAAa,GACS,EAAe,EAAE,CAAC,CACxC,oBAAC,kBAAkB;IACjB,oBAAC,kCAAkC;QACjC,oBAAC,+BAA+B,QAAE,KAAK,CAAmC;QAC1E,oBAAC,yBAAyB,QAAE,QAAQ,CAA6B;QACjE,oBAAC,MAAM,QAAE,aAAa,CAAU,CACG;IACrC,oBAAC,gCAAgC,QAAE,KAAK,CAAoC,CACzD,CACtB,CAAC"}