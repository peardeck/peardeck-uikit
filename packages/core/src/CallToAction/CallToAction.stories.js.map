{"version":3,"file":"CallToAction.stories.js","sourceRoot":"","sources":["CallToAction.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAC1B,OAAO,EAAE,SAAS,EAAE,MAAM,kBAAkB,CAAC;AAC7C,OAAO,EAAE,IAAI,EAAE,SAAS,EAAE,MAAM,wBAAwB,CAAC;AACzD,OAAO,EAAE,YAAY,EAAE,MAAM,gBAAgB,CAAC;AAE9C,SAAS,CAAC,mBAAmB,EAAE,MAAM,CAAC;KACnC,YAAY,CAAC,SAAS,CAAC;KACvB,GAAG,CAAC,SAAS,EAAE,GAAG,EAAE,CAAC,CACpB,oBAAC,YAAY,IACX,aAAa,EAAE,IAAI,CAAC,aAAa,EAAE,qBAAqB,CAAC,EACzD,KAAK,EAAC,iBAAiB,EACvB,KAAK,EAAE,IAAI,CAAC,OAAO,EAAE,qBAAqB,CAAC,IAE1C,IAAI,CACH,SAAS,EACT,8HAA8H,CAC/H,CACY,CAChB,CAAC,CAAC"}