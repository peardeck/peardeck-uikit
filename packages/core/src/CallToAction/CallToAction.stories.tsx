import React from "react";
import { storiesOf } from "@storybook/react";
import { text, withKnobs } from "@storybook/addon-knobs";
import { CallToAction } from "./CallToAction";

storiesOf("core/CallToAction", module)
  .addDecorator(withKnobs)
  .add("Default", () => (
    <CallToAction
      buttonContent={text("Button Text", "Default Button Text")}
      image="Image goes here"
      title={text("Title", "Default Header Text")}
    >
      {text(
        "Content",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque est felis, cursus efficitur aliquet ac, ullamcorper a diam."
      )}
    </CallToAction>
  ));
