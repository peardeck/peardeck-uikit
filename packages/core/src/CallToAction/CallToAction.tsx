import React, { ReactNode } from "react";
import styled from "styled-components";
import { Button } from "../Button/Button";

const StyledCallToAction = styled.div`
  display: inline-flex;
  flex-direction: row;
  flex-grow: 1;
  align-items: center;
  justify-content: space-between;
  background-color: #edf0f4;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.08);
  border-radius: 0.375em;
  transition: all 0.25s ease;
  transform: translateZ(0px);

  :hover {
    transform: translateY(-2px);
    box-shadow: 0px 5px 3px 0px rgba(0, 0, 0, 0.08);
  }
`;

const StyledCallToActionContentContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin: 1em;
  max-width: 20em;
`;

const StyledCallToActionContent = styled.p`
  flex: 1;
  margin: 1em 0;
`;

const StyledCallToActionContentHeader = styled.h3`
  margin: 0.25em 0;
`;

const StyledCallToActionImageContainer = styled.div`
  background-color: #999;
  margin: 1em;
  padding: 1em;
`;

interface CallToActionPropTypes {
  children: ReactNode;
  title: ReactNode;
  image: ReactNode;
  buttonContent: ReactNode;
}

export const CallToAction = ({
  children,
  title,
  image,
  buttonContent,
}: CallToActionPropTypes): JSX.Element => (
  <StyledCallToAction>
    <StyledCallToActionContentContainer>
      <StyledCallToActionContentHeader>{title}</StyledCallToActionContentHeader>
      <StyledCallToActionContent>{children}</StyledCallToActionContent>
      <Button>{buttonContent}</Button>
    </StyledCallToActionContentContainer>
    <StyledCallToActionImageContainer>{image}</StyledCallToActionImageContainer>
  </StyledCallToAction>
);
