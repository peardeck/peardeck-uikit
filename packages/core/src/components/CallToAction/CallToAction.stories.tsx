import React from "react";
import { text } from "@storybook/addon-knobs";
import { CallToAction } from "./CallToAction";

export default {
  component: CallToAction,
  title: "core|CallToAction",
};

export const Default = () => (
  <CallToAction
    buttonContent={text("Button Text", "Do it now!")}
    image="Image goes here"
    title={text("Title", "Now is the time to act!")}
  >
    {text(
      "Content",
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque est felis, cursus efficitur aliquet ac, ullamcorper a diam."
    )}
  </CallToAction>
);

export const WithoutAnImage = () => (
  <CallToAction
    buttonContent={text("Button Text", "Do it now!")}
    title={text("Title", "Now is the time to act!")}
  >
    {text(
      "Content",
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque est felis, cursus efficitur aliquet ac, ullamcorper a diam."
    )}
  </CallToAction>
);
