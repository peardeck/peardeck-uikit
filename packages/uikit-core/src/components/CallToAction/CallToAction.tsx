import React, { ReactNode } from "react";
import { Box } from "../Box/Box";
import { Card, CardProps } from "../Card/Card";
import { Button } from "../Button/Button";
import { Heading2 } from "../Heading/Heading";

export type CallToActionProps = CardProps & {
  children?: ReactNode;
  title: ReactNode;
  image?: ReactNode;
  buttonContent: ReactNode;
};

export const CallToAction = ({
  children,
  title,
  image,
  buttonContent,
}: CallToActionProps): JSX.Element => (
  <Card p={2} display="inline-flex">
    <Box display="inline-flex" flexDirection="column" alignItems="flex-start">
      <Heading2 paddingLeft={1} marginBottom={2}>
        {title}
      </Heading2>
      <Box paddingLeft={1} marginTop={2} marginBottom={1}>
        {children}
      </Box>
      <Button
        colorVariant="accent"
        marginTop={1}
        borderRadius={4}
        sizeVariant="large"
      >
        {buttonContent}
      </Button>
    </Box>
    {image && (
      <Card marginLeft={2} depth={0}>
        {image}
      </Card>
    )}
  </Card>
);

CallToAction.displayName = "CallToAction";
