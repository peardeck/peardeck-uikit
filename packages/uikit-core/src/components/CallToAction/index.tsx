export { CallToAction, CallToActionProps } from "./CallToAction";
