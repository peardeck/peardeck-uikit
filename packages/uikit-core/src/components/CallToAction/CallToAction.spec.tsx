import React from "react";
import { create } from "react-test-renderer";
import { CallToAction } from "./CallToAction";

describe("CallToAction", () => {
  it("matches the snapshot", () => {
    const json = create(
      <CallToAction
        title="Call To Action!"
        buttonContent="Do the thing"
        image="foo"
      />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it("renders with out an image", () => {
    const json = create(
      <CallToAction title="Call To Action!" buttonContent="Do the thing" />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });
});
