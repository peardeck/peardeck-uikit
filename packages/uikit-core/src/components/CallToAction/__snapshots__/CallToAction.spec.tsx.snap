// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToAction matches the snapshot 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  padding-left: 4px;
  margin-top: 8px;
  margin-bottom: 4px;
}

.c0 {
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px;
  box-shadow: 0 1px 4px 0 rgba(0,0,0,0.27);
  border-radius: 3px;
}

.c5 {
  border-radius: 3px;
  border: 1px solid #6b6b6b;
  margin-left: 8px;
  padding: 4px;
  box-shadow: none;
  border-radius: 3px;
  border: 1px solid #6b6b6b;
}

.c4 {
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  font-size: 24px;
  padding-left: 32px;
  padding-right: 32px;
  padding-top: 16px;
  padding-bottom: 16px;
  border-radius: 24px;
  border-width: 2px;
  border-color: #ffffff;
  color: #ffffff;
  background-color: #d24949;
  box-shadow: 0 1px 4px 0 rgba(0,0,0,0.27);
  margin-top: 4px;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  text-align: center;
}

.c4 > * {
  vertical-align: middle;
}

.c4:disabled {
  opacity: 0.3;
}

.c4:hover {
  box-shadow: 0 2px 8px 0 rgba(0,0,0,0.27);
  background-color: #ba2e2e;
}

.c4:active {
  box-shadow: rgba(0,0,0,0.12) 0px 0px 1px inset;
}

.c2 {
  padding-left: 4px;
  margin-bottom: 8px;
  margin: 0;
}

<div
  className="c0"
  display="inline-flex"
>
  <div
    className="c1"
    display="inline-flex"
  >
    <h2
      className="c2"
    >
      Call To Action!
    </h2>
    <div
      className="c3"
    />
    <button
      className="c4"
      color="#ffffff"
      fontFamily="inherit"
      fontWeight="bold"
    >
      Do the thing
    </button>
  </div>
  <div
    className="c5"
  >
    foo
  </div>
</div>
`;

exports[`CallToAction renders with out an image 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  padding-left: 4px;
  margin-top: 8px;
  margin-bottom: 4px;
}

.c0 {
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px;
  box-shadow: 0 1px 4px 0 rgba(0,0,0,0.27);
  border-radius: 3px;
}

.c4 {
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  font-size: 24px;
  padding-left: 32px;
  padding-right: 32px;
  padding-top: 16px;
  padding-bottom: 16px;
  border-radius: 24px;
  border-width: 2px;
  border-color: #ffffff;
  color: #ffffff;
  background-color: #d24949;
  box-shadow: 0 1px 4px 0 rgba(0,0,0,0.27);
  margin-top: 4px;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  text-align: center;
}

.c4 > * {
  vertical-align: middle;
}

.c4:disabled {
  opacity: 0.3;
}

.c4:hover {
  box-shadow: 0 2px 8px 0 rgba(0,0,0,0.27);
  background-color: #ba2e2e;
}

.c4:active {
  box-shadow: rgba(0,0,0,0.12) 0px 0px 1px inset;
}

.c2 {
  padding-left: 4px;
  margin-bottom: 8px;
  margin: 0;
}

<div
  className="c0"
  display="inline-flex"
>
  <div
    className="c1"
    display="inline-flex"
  >
    <h2
      className="c2"
    >
      Call To Action!
    </h2>
    <div
      className="c3"
    />
    <button
      className="c4"
      color="#ffffff"
      fontFamily="inherit"
      fontWeight="bold"
    >
      Do the thing
    </button>
  </div>
</div>
`;
